recaptcha/recaptcha-enterprise/src/main/kotlin/com/pingidentity/recaptcha/enterprise/RecaptchaClientProvider.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2025 Ping Identity Corporation. All rights reserved.
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package com.pingidentity.recaptcha.enterprise
+
+import android.app.Application
+import com.google.android.recaptcha.Recaptcha
+import com.google.android.recaptcha.RecaptchaAction
+import com.google.android.recaptcha.RecaptchaClient
+
+/**
+ * Provider interface for managing ReCaptcha Enterprise client operations.
+ *
+ * This interface abstracts the creation and management of RecaptchaClient instances
+ * and the execution of ReCaptcha actions. It allows for custom implementations
+ * to provide different ReCaptcha handling strategies, testing mocks, or enhanced
+ * functionality while maintaining a consistent API.
+ *
+ * Implementations should handle:
+ * - Client creation and configuration
+ * - Action execution with proper timeout handling
+ * - Error handling and recovery
+ * - Resource cleanup when necessary
+ */
+interface RecaptchaClientProvider {
```

Comment:
I don't think we need RecaptchaClientProvider interface, the interface fetchClient returning a RecaptchaClient and the execute function just call the execute function inside the RecaptchaCLient. I would remove the RecaptchaClientProvider and DefaultRecaptchaClientProvider. I don't see a options to provide different implementation of ```RecaptchaClient```
If we want to provide custom ```RecaptchaClient``` I would add another attribute under config with 
```kotlin
    var recaptchaClient: (String) -> RecaptchaClient = {}
```